libmcount/mcount.c

```diff
@@ -1716,6 +1723,133 @@ static void mcount_script_init(enum uftrace_pattern_type patt_type)
 	strv_free(&info.cmds);
 }
 
+int agent_init(struct sockaddr_un *addr)
+{
+	int sfd;
+
+	agent_run = true;
+
+	if (mkdir(MCOUNT_AGENT_SOCKET_DIR, 0775) == -1) {
+		if (errno != EEXIST) {
+			pr_warn("error creating run directory %s", MCOUNT_AGENT_SOCKET_DIR);
+		}
+	}
+
+	sfd = create_socket(addr, getpid());
+	if (sfd == -1)
+		return sfd;
+
+	if (unlink(addr->sun_path) == -1) {
+		if (errno != ENOENT) {
+			pr_warn("cannot unlink %s", addr->sun_path);
+		}
+	}
+
+	if (listen_socket(sfd, addr))
+		return -1;
+
+	return sfd;
+}
+
+void agent_fini(int sfd)
+{
+	if (sfd != -1)
+		close(sfd);
+}
+
+/* Agent routine, applying instructions from the CLI. */
+void *apply_client_commands(void *arg)
```

Comment:
It'd be nice we could have `agent_` prefix.  How about `agent_apply_commands()`?